pkg/utils/k8sutil.go

```diff
@@ -372,6 +372,17 @@ func EnsureK8sResources(ns, name string, funcObj *spec.Function, client kubernet
 		addInitContainerAnnotation(dpm)
 	}
 
+	// update deployment for memory request
+	if funcObj.Spec.Memory != "" {
+		//mem value has already been validated before
+		quantity, _ := resource.ParseQuantity(funcObj.Spec.Memory)
```

Comment:
don't re-parse values on the server-side (and if you do, don't throw away the error!), just store the "parsed" quantity in the function Spec in the first place.  ie: arrange it so that the code here becomes a simple value assignment.